### PR TITLE
Refactor ModalBase from existing modals

### PR DESCRIPTION
**Pull request by anthonyshibitov**
Refactor ModalBase from existing modals

ModalBase can accept an imageModal prop to change behavior and match existing functionality.

### Contributor checklist

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

Refactor modal functionality to a separate `ModalBase.vue` component.

`imageBase` can be added to the `<ModalBase>` component to mimic the functionality we currently have, where images and text modals behave (click image to close modal) and display (text modal has close button and title in a card, with the content within it) differently.

Note that this currently does not support i8n in the modal title (specifically, `{{ $t("components.modal-qr-code.header") }}`). I wanted to get some feedback on this before fleshing out the remaining portions, to make sure this design is what we're looking for. Let me know if this is okay, and I will commit another change to try and fix the i8n issue (I'm currently not sure how to do this, any ideas would be great lol 😀). If we want the modal to behave differently, I'll go back to the drawing board.

Also, note to myself, the bug where clicking to the left or right of the image modal does not close the modal still exists 🙃

### Related issue

- #582

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks @anthonyshibitov! I'll check this in the coming days and provide some feedback 😊

**Comment by andrewtavis** [comment]
Looking good so far, @anthonyshibitov! I'm going to wait on #565 to get merged before bringing this in. Some thoughts on this:

- For the image maybe removing the active state when it's expanded would be good as when the user clicks the image to close it gets a border around it, but clicking the sides it won't (maybe will be solved by the change to get the left and right parts of the expanded modal to work to close the image)
- Looks like you're using npm to set things up rather than yarn, which is what it is, but I'd warn against installing any dependencies in this way as we've found that npm installed dependencies can break things
  - Regardless, can you revert the package-lock.json files 🙃 
- Re you're question on the i18n changes to this:
  - I think this could make sense, or maybe we can also just drop all the card contents including the title into s `<slot>`? Might be a bit weird with the close icon, but that way all the logic for the modal card would be in the slot and we wouldn't also have to assign some parts in the props.
  - Lemme know what you think! It'll work either way :)

**Comment by andrewtavis** [comment]
I can take a look at it myself, but when I click the image to close it there's a border that appears around it, whereas when we'd click the sides there wouldn't be. Trying to make sure that the closing modal experience is the same :)

**Comment by andrewtavis** [comment]
Are we good for a final review on this, @anthonyshibitov? :)

**Comment by anthonyshibitov** [comment]
> Are we good for a final review on this, @anthonyshibitov? :)

yessir!

**Comment by andrewtavis** [comment]
Wunderbar 😊 I'll get to it soon then :)

File: frontend/components/modal/ModalImage.vue [hunk 1]
```diff
@@ -1,75 +1,42 @@
...
-            class="object-contain p-12"
-            :src="imageURL + '_dark.png'"
-            :alt="$t(imageAltText)"
-          />
-        </button>
-      </DialogPanel>
-    </div>
-  </Dialog>
+  <ModalBase imageModal>
+    <template #normalDisplay>
```
Comment (andrewtavis) [inline frontend/components/modal/ModalImage.vue:3]: This is some great stuff here, @anthonyshibitov :)

Comment (andrewtavis) [reply frontend/components/modal/ModalImage.vue:3]: Really well organized. The double multiple slots that are named makes this super easy to follow 😊

**Review by andrewtavis — APPROVED** [review]
Thanks for this, @anthonyshibitov! I'll try to play around to get the image boundaries to be clickable to close it :)

**Pull request merged**